src/main/java/frc/robot/subsystems/led/Led2023.java

```diff
@@ -1,50 +1,60 @@
+import frc.robot.commands.arm.ArmCalibrateZeroAtHomeCMD;
 import frc.robot.commands.arm.ArmFloorCMD;
 import frc.robot.commands.arm.ArmScoreHighNodeCMD;
 import frc.robot.commands.arm.ArmScoreMidNodeCMD;
 import frc.robot.commands.arm.ArmShelfCMD;
+import frc.robot.commands.auto.complex.OnlyBackup;
+import frc.robot.commands.auto.complex.OnlyScore;
+import frc.robot.commands.auto.complex.ScoreAndBackUp;
 import frc.robot.commands.intakerelease.HoldCMD;
 import frc.robot.commands.intakerelease.IntakeAndRaise;
 import frc.robot.commands.intakerelease.IntakeCMD;
 import frc.robot.commands.intakerelease.ReleaseCMD;
 import frc.robot.subsystems.arm.Arm;
+import frc.robot.subsystems.drive.Drive;
 import frc.robot.subsystems.intakerelease.IntakeRelease;
+import org.littletonrobotics.junction.networktables.LoggedDashboardChooser;
 
 public class Led2023 extends SubsystemBase {
   public DoubleLEDStrip ledStrip;
-  private final boolean USE_BATTERY_CHECK = true;
-  private final double BATTER_MIN_VOLTAGE = 9.0; // 10.0
-  private final boolean CHECK_ARM_CALIBRATION = true;
-  private boolean FINISHED_RAINBOW_ONCE = false;
-
-  private final double SHOOTING_TIMER_SPEED = 0.1;
-  private final double RAINBOW_TIMER_SPEED = 0.02;
-  private final int RAINBOW_AMOUNT = 20;
-  private IntakeRelease intakerelease;
-  private Arm arm;
 
-  private double color = 0;
-  private Timer rainbowTimer = new Timer();
-  private Timer purpleTimer = new Timer();
-  protected double lastLoopTime = 0;
+  private Timer balanceTimer = new Timer();
   private Timer defaultTimer = new Timer();
+  private ColorScheme lastColorScheme;
+  private boolean doneBalancing = false;
+  private boolean finishedRainbowOnce = false;
+
+  private static final boolean USE_BATTERY_CHECK = true;
+  private static final double BATTER_MIN_VOLTAGE = 9.0;
+  private static final boolean CHECK_ARM_CALIBRATION = true;
+  private static final double AFTER_CALIBRATED_CLRSCM = 2.5;
+  private static final double MAX_ANGLE_VELOCITY_DEGREES_PER_SECOND = 8.0;
+  private static final COLORS_467 BATTERY_CHECK_COLOR = COLORS_467.Orange;
+  private static final COLORS_467 ARM_UNCALIBRATED_COLOR = COLORS_467.Red;
 
-  public static final double TARGET_MAX_RANGE = 100.0;
-  public static final double TARGET_MAX_ANGLE = 15.0;
-  public static final double BALL_MAX_RANGE = 100.0;
-  public static final double BALL_MAX_ANGLE = 15.0;
-  private COLORS_467 batteryCheckColor = COLORS_467.Orange;
+  private IntakeRelease intakerelease;
+  private Arm arm;
+  private Drive drive;
+  private LoggedDashboardChooser<Command> autoChooser;
+
+  private VictoryLeds balanceVictoryLeds = new VictoryLeds(COLORS_467.Blue, COLORS_467.Gold);
+  private VictoryLeds scoreVictoryLeds = new VictoryLeds(COLORS_467.Yellow, COLORS_467.Purple);
+  private Rainbows rainbowLed = new Rainbows();
+  private Patterns colorPatterns = new Patterns();
+  private SetThirdLeds setOneThird = new SetThirdLeds();
```

Comment:
This should be named `setThirdLeds` to be more like the class name rather than matching the name of the method that get's called later, which is confusing.